mongo/integration/unified/client_entity.go

```diff
@@ -53,19 +55,17 @@ func newClientEntity(ctx context.Context, em *EntityMap, entityOptions *entityOp
 	}
 	// If not observing sensitive commands, add security-sensitive commands
 	// to ignoredCommands by default.
-	if entityOptions.ObserveSensitiveCommands != nil && !*entityOptions.ObserveSensitiveCommands {
+	if entityOptions.ObserveSensitiveCommands == nil || !*entityOptions.ObserveSensitiveCommands {
```

Comment:
Oh good catch!

Reply:
Yeah this boolean logic was incorrect before. Previously it checked if `ObserveSensitiveCommands` was set and was false. But we really want to check if `ObserveSensitiveCommands` is unset OR it's set and is false.